core/src/main/java/io/ddf/etl/ATimeSeriesHandler.java

```diff
@@ -0,0 +1,127 @@
+package io.ddf.etl;
+
+
+import io.ddf.DDF;
+import io.ddf.analytics.ABinningHandler.BinningType;
+import io.ddf.exception.DDFException;
+import io.ddf.misc.ADDFFunctionalGroupHandler;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import com.google.common.collect.Lists;
+
+public abstract class ATimeSeriesHandler extends ADDFFunctionalGroupHandler implements IHandleTimeSeries {
+
+  protected String mTimestampColumn;
+  protected String mTsIDColumn = null;
+
+
+  public ATimeSeriesHandler(DDF theDDF) {
+    super(theDDF);
+
+  }
+
+  public void setTimeStampColumn(String colName) {
+    mTimestampColumn = colName;
+  }
+
+  public String getTimeStampColumn() {
+    return mTimestampColumn;
+  }
+
+
+  public String getTsIDColumn() {
+    return mTsIDColumn;
+  }
+
+  public void setTsIDColumn(String colName) {
+    this.mTsIDColumn = colName;
+  }
+
+  @Override
+  public DDF downsample(String timestampColumn, List<String> aggregateFunctions, int interval, TimeUnit timeUnit)
+      throws DDFException {
+
+    this.mTimestampColumn = timestampColumn;
+    List<String> groupByCols = Lists.newArrayList(timestampColumn);
+    if (mTsIDColumn != null && !mTsIDColumn.isEmpty()) {
+      groupByCols.add(mTsIDColumn);
```

Comment:
`mTsIDColumn` shouldn't be added here, this method should only downsample based on  `timestampColumn`.